code/modules/mob/living/inhand_holder.dm

```diff
@@ -22,6 +21,7 @@
 	if(worn_slot_flags)
 		slot_flags = worn_slot_flags
 	deposit(M)
+	. = ..()
```

Comment:
```suggestion
	. = ..()
	deposit(M)
```
We want to call parent as soon as possible during initializations.

Reply:
this wouldn't actually fix the bug, because the `icon_state` is updated via the `deposit()` proc

Reply:
aha, I see it now